core/src/types/eth_abi.rs

```diff
@@ -74,10 +66,16 @@ pub trait Encode<const N: usize>: Sized {
 
     /// Encodes a slice of [`Token`] instances, and returns the
     /// keccak hash of the encoded string appended to an Ethereum
-    /// signature header.
-    fn signed_keccak256(&self) -> KeccakHash {
+    /// signature header. This can then be signed.
+    fn signable_keccak256(&self) -> KeccakHash {
+        let mut output = [0; 32];
         let message = self.encode().into_inner();
-        SignedKeccakAbi::as_signable(&message)
+        let signable_bytes = SignedAbiBytes::as_signable(&message);
+        let mut state = Keccak::v256();
+        state.update(&signable_bytes);
+        state.finalize(&mut output);
```

Comment:
https://ethereum.stackexchange.com/a/30173
this is incorrect I think. the header should be prepended to the message to be signed. here you are hashing the header as well. it should be something like `sign(header || keccak(arbitrary_data))`

Reply:
also maybe rename `SignedAbiBytes` to `SignableAbiBytes`, for consistency. perhaps even `SignableEthBytes`, since the data doesn't strictly have to be ABI encoded anymore